File: 2023 yard hoof cup/Python The end scene/8.py
def change(it: str) -> str:
    if it[0] >= '0' and it[0] <= '9':
        num = int(it)
        res = ""
        while num > 0:
            res += str(chr(num % 26 + ord('a')))
            num //= 26
        return res[::-1]
    else:
        num = 0
        bas = 1
        for i in range(len(it) - 1, -1, -1):
            num += bas * (ord(it[i]) - ord('a'))
            bas *= 26
        return str(num)


def next(i: int, A: str) -> int:
    p1 = i
    p2 = i
    while p1 < len(A) and A[p1] >= '0' and A[p1] <= '9':
        p1 += 1
    while p2 < len(A) and A[p2] >= 'a' and A[p2] <= 'z':
        p2 += 1
    return max(p1, p2)


def solve():
    global A

    A = input()
    i = 0
    res = ""
    while i < len(A):
        j = next(i, A)
        res += change(A[i:j])
        i = j
    print(res)


if __name__ == '__main__':
    solve()

File: 2023 yard hoof cup/C++ The end scene/8.cpp
#include <bits/stdc++.h>

using namespace std;
#define int long long

string A;

void change(string&it) {
    if (it[0] >= '0' && it[0] <= '9') {
        int num = stoll(it);
        string res;
        for (; num > 0; num /= 26) {
            res += char(num % 26 + 'a');
        }
        reverse(res.begin(), res.end());
        it = res;
    } else {
        int num = 0;
        int bas = 1;
        for (int i = (int) it.size() - 1; i >= 0; i--, bas *= 26) {
            num += bas * (it[i] - 'a');
        }
        it = to_string(num);
    }
}

int next(int i) {
    int p1;
    int p2;
    for (p1 = i; p1 < A.size() && A[p1] >= '0' && A[p1] <= '9'; p1++) {
    }
    for (p2 = i; p2 < A.size() && A[p2] >= 'a' && A[p2] <= 'z'; p2++) {
    }
    return max(p1, p2);
}

void solve() {
    cin >> A;
    string res;
    for (int i = 0; i < A.size(); i = next(i)) {
        int j = next(i);
        string temp = A.substr(i, j - i);
        change(temp);
        res += temp;
    }
    cout << res << endl;
}

auto main() -> signed {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);

    solve();
    return 0;
}
